spec/javascripts/Dashboard/components/PotentialUpgradesWidget.spec.jsx

```diff
@@ -0,0 +1,61 @@
+// @flow
+
+import React from 'react'
+
+import { PotentialUpgradesWidget, Props } from 'Dashboard/components/PotentialUpgradesWidget'
+import { mount } from 'enzyme'
+
+const defaultProps: Props = {
+  violations: [
+    {
+      id: 1,
+      account_id: 1,
+      account_name: 'Account Name',
+      alerts_count: 10,
+      url: '/apiconfig/alerts?search%5Baccount_id%5D=1&search%5Blevel%5D=100&search%5Btimestamp%5D%5B%5D=2022-09-24&search%5Btimestamp%5D%5B%5D=2022-10-24'
+    }
+  ],
+  incorrectSetUp: true,
+  links: {
+    adminServiceApplicationPlans: '/apiconfig/services/1/application_plans',
+    settingsAdminService: '/apiconfig/services/1/settings#web_provider'
+  }
+}
+
+const mountWrapper = (props: Props) => mount(<PotentialUpgradesWidget {...{ ...defaultProps, ...props }}/>)
+
+it('should render', () => {
+  const wrapper = mountWrapper(defaultProps)
+
+  expect(wrapper.exists()).toBe(true)
+})
+
+describe('when the setup is incorrect', () => {
+  it('should render incorrect setup messages', () => {
+    const wrapper = mountWrapper(defaultProps)
+
+    expect(wrapper.text().includes('In order to show Potential Upgrades, add 1 or more usage limits to')).toBe(true)
+    expect(wrapper.text().includes('Web Alerts for Admins of this Account of 100%')).toBe(true)
+  })
+})
+
+describe('when the setup is correct', () => {
+  describe('and there are violations', () => {
+    it('should render a list of violations', () => {
+      let props: Props = { ...defaultProps, incorrectSetUp: false }
```

Comment:
As said in the previous file, I suggest changing `let` for `const` since these props won't be changing:

```suggestion
      const props: Props = { ...defaultProps, incorrectSetUp: false }
```

Reply:
Updated in bf00c23.